sound/soc/soc-topology.c

```diff
@@ -519,10 +519,20 @@ static void remove_widget(struct snd_soc_component *comp,
 		/* enumerated widget mixer */
 		for (i = 0; w->kcontrols != NULL && i < w->num_kcontrols; i++) {
 			struct snd_kcontrol *kcontrol = w->kcontrols[i];
-			struct soc_enum *se =
-				(struct soc_enum *)kcontrol->private_value;
+			struct soc_enum *se;
 			int j;
 
+			/* in some case:tplg incorrect configuration.
+			 * the kcontrol{} is not allocated successfully
```

Comment:
I would change both comments to 
/* kcontrol not guaranteed to be created so validate */

Reply:
OK. I will update it. thanks